src/mongo/MongoCommands.ts

```diff
@@ -18,16 +18,61 @@ import { MongoFindOneResultEditor } from './editors/MongoFindOneResultEditor';
 import { MongoCommand } from './MongoCommand';
 import { MongoDatabaseTreeItem } from './tree/MongoDatabaseTreeItem';
 
+const output = vscodeUtil.getOutputChannel();
+const notInScrapbookMessage = "This command can only be run inside of a MongoDB scrapbook (*.mongo)";
+
 export class MongoCommands {
+
+	public static async executeAllCommandsFromActiveEditor(database: IAzureParentNode<MongoDatabaseTreeItem>, extensionPath, editorManager: CosmosEditorManager, tree: AzureTreeDataProvider, context: IActionContext): Promise<void> {
+		output.show();
+		output.appendLine("Running all commands in scrapbook...")
+		let commands = MongoCommands.getAllCommandsFromActiveEditor();
+		await MongoCommands.executeCommands(vscode.window.activeTextEditor, database, extensionPath, editorManager, tree, context, commands);
+	}
+
 	public static async executeCommandFromActiveEditor(database: IAzureParentNode<MongoDatabaseTreeItem>, extensionPath, editorManager: CosmosEditorManager, tree: AzureTreeDataProvider, context: IActionContext): Promise<void> {
+		const commands = MongoCommands.getAllCommandsFromActiveEditor();
 		const activeEditor = vscode.window.activeTextEditor;
-		if (activeEditor.document.languageId !== 'mongo') {
-			vscode.window.showInformationMessage("This command can only be run inside of a MongoDB scrapbook (*.mongo)");
-			return;
-		}
 		const selection = activeEditor.selection;
-		const command = MongoCommands.getCommand(activeEditor.document.getText(), selection.start);
+		const command = MongoCommands.findCommandAtPosition(commands, selection.start);
+		return this.executeCommand(activeEditor, database, extensionPath, editorManager, tree, context, command);
+	}
+
+	public static async executeCommandFromText(database: IAzureParentNode<MongoDatabaseTreeItem>, extensionPath, editorManager: CosmosEditorManager, tree: AzureTreeDataProvider, context: IActionContext, commandText: string): Promise<void> {
+		const activeEditor = vscode.window.activeTextEditor;
+		const command = MongoCommands.getCommandFromText(commandText, new vscode.Position(0, 0));
+		return this.executeCommand(activeEditor, database, extensionPath, editorManager, tree, context, command);
+	}
+
+	public static getAllCommandsFromActiveEditor(): MongoCommand[] {
+		const activeEditor = vscode.window.activeTextEditor;
+		if (activeEditor) {
+			const commands = MongoCommands.getAllCommandsFromTextDocument(activeEditor.document);
+			return commands;
+		} else {
+			throw new Error(notInScrapbookMessage);
```

Comment:
This is currently there (not my code) because the commands are exposed in the command palette but are always enabled. Excluding them sounds good, except that would make it inaccessible unless I add a keyboard shortcut.

Reply:
I thought it was accessible through code lens?

Reply:
"Accessible" accessible - ie through keyboard. Maybe code lens does have this, will have to look it up.  If there is, I have no strong opinion.

Reply:
Can't seem to find a shortcut for this in vscode.

Reply:
Oh that "Accessible". Well I definitely think you should exclude `executeMongoCommand` from the command palette. You lose context of the cursor when you open the command palette so it always runs the first command. Plus it already has a keyboard shortcut.

I guess it might make sense to have `executeAllMongoCommands` in the command palette, but a keyboard shortcut feels more natural to me. No strong opinion on this one. I will point out that whether or not you add a default keyboard shortcut, the user can add a custom keyboard shortcut for literally any command we define.

Reply:
@fiveisprime any suggestions on a keyboard shortcut for execute all?  
Currently CTRL+SHIFT+” is execute.  I would choose CTRL+SHIFT+ENTER but that’s Insert Line Above”.